pytorch_lightning/metrics/functional/regression.py

```diff
@@ -279,17 +279,20 @@ def ssim(
     device = pred.device
 
     channel = pred.size(1)
+
     kernel = _gaussian_kernel(channel, kernel_size, sigma, device)
-    mu_pred = F.conv2d(pred, kernel, groups=channel)
-    mu_target = F.conv2d(target, kernel, groups=channel)
 
-    mu_pred_sq = mu_pred.pow(2)
-    mu_target_sq = mu_target.pow(2)
-    mu_pred_target = mu_pred * mu_target
+    input_list = torch.cat([pred, target, pred * pred, target * target, pred * target])
```

Comment:
can you add some comments for others to enhance readability and make it easier to understand your code?

Reply:
Got it, justus